### ISSUE TITLE
Render API endpoint results in a 500 server error.

### ISSUE THREAD

**Issue by eugeneingram** [VirtusLab/strapi-plugin-navigation#44]
Having constructed a simple navigation using the UI builder (2 menu items):

![image](https://user-images.githubusercontent.com/11439060/109875117-ef1c0a00-7c67-11eb-9249-aa353d674d2b.png)

I'm able to view this menu output on the `/navigation/1` API endpoint:

![image](https://user-images.githubusercontent.com/11439060/109875180-04913400-7c68-11eb-9162-a33f9308c7a8.png)

However, attempting to user the render endpoint `/navigation/render/1?type=rfr` gives a 500 response error:

![image](https://user-images.githubusercontent.com/11439060/109875251-212d6c00-7c68-11eb-9216-60fbd997de0e.png)

**Comment by eugeneingram** [comment]
If it helps, digging in further I think that the issue is related to my content items being singular content types.  You addressed this requirement as part of issue #23 here: https://github.com/VirtusLab/strapi-plugin-navigation/issues/23

If I make a two item menu with a collection content type it works fine when rendering.

**Comment by cyp3rius** [comment]
Hi @eugeneingram ,

May I ask you for sharing at least one Single Content Type definition file (you can drop all attributes to not expose any protected value)? Also does those Single Types being part of Draft & Publish flow? If yes, have they being published?

**Comment by eugeneingram** [comment]
`{
  "kind": "singleType",
  "collectionName": "page_homes",
  "info": {
    "name": "Page Home"
  },
  "options": {
    "increments": true,
    "timestamps": true,
    "draftAndPublish": true
  },
  "attributes": {
    "Subtitle": {
      "type": "string"
    },
    "Title": {
      "type": "string"
    },
    "Summary": {
      "type": "text"
    },
    "Image": {
      "model": "file",
      "via": "related",
      "allowedTypes": [
        "images"
      ],
      "plugin": "upload",
      "required": false
    },
    "Components": {
      "type": "dynamiczone",
      "components": [
        ...
      ]
    },
    "navigation": {
      "model": "navigationitem",
      "plugin": "navigation",
      "via": "related",
      "configurable": false,
      "hidden": true
    }
  }
}`

**Comment by eugeneingram** [comment]
Hi @cyp3rius 

Thanks for coming back on this.  I think you were on to something with the Publish status of the Single content types.  I added them to the navigation whilst they remained on published.  This leads directly to the 500 error if you add even one menu item to a piece of unpublished content.

Strangely.  If I delete all menu items, publish both the single content types and add them to the menu again, the render function works as expected.  Even if I now unpublish both single content items, the render function continues to work.  So there is something unusual going on here.

At least now I can build a navigation menu with a working render function so that is great.  I think it is just a little fragile if someone accidentally adds a menu item to some unpublished content.  It will break the entire navigation if the render API endpoint is being used.

**Comment by cyp3rius** [comment]
Agreed, this thing with unpublished content is currently in development, expecting to be done around the weekend :) Will try to sort out also all issues you raised here.

**Comment by eugeneingram** [comment]
Wonderful, thanks for your help.  Thanks also for the plugin.  It's very handy!

**Comment by cyp3rius** [comment]
@eugeneingram new version that contains fix has been released as `1.0.0-beta.17`. Test an close the ticket please.
